airbyte-webapp/src/components/JobItem/components/MainInfo.tsx

```diff
@@ -37,25 +38,55 @@ interface MainInfoProps {
 
 const MainInfo: React.FC<MainInfoProps> = ({ job, attempts = [], isOpen, onExpand, isFailed }) => {
   const jobStatus = getJobStatus(job);
+  const jobConfigType = getJobConfig(job);
+  const streamsToReset = "job" in job ? job.job.resetConfig?.streamsToReset : undefined;
   const isPartialSuccess = partialSuccessCheck(attempts);
 
   const statusIcon = () => {
     switch (true) {
       case jobStatus === JobStatus.cancelled:
+      case !isPartialSuccess && isFailed:
         return <StatusIcon status="error" />;
       case jobStatus === JobStatus.running:
         return <StatusIcon status="loading" />;
       case jobStatus === JobStatus.succeeded:
         return <StatusIcon status="success" />;
       case isPartialSuccess:
         return <StatusIcon status="warning" />;
-      case !isPartialSuccess && isFailed:
-        return <StatusIcon status="error" />;
       default:
         return null;
     }
   };
 
+  const label = () => {
+    let status = "";
+    switch (true) {
```

Comment:
I'm not a fan of [switch (true)](https://medium.com/trabe/using-switch-true-in-javascript-986e8ad8ae4f) as a pattern. I would prefer we do something else.
This should be a useMemo.